### ISSUE TITLE
typescript/lib/lib.d.ts not found in all jsconfig.json files without target

### ISSUE THREAD

**Issue by 06000208** [microsoft/vscode#168420]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.74.0 and 1.75.0-insider
- OS Version: Windows 11 x64 10.0.22621

Every project on my local system containing a jsconfig.json file without a target property has started reporting this error:

```
File 'd:/Program Files/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/lib.d.ts' not found.
  The file is in the program because:
    Default library for target 'es3'
```

Some projects have it multiple times, others only once. I freshly installed insiders to test, with no settings or extensions, and it occurs there too.

It doesn't occur if you have `target` as one of your compiler options, but this wasn't necessary before. My code isn't compiled with typescript; I use jsconfig so vscode recognizes my files as being part of the same project, and other similar reasons.

Working as normal when target is specified:
![2022-12-08-Code_-_Insiders-1670492789](https://user-images.githubusercontent.com/52764066/206414914-a78dc4ef-c9dc-43b3-bf0b-11637f4e4622.png)

Examples of the issue:
![2022-12-08-Code_-_Insiders-1670491612](https://user-images.githubusercontent.com/52764066/206411622-db1bdad9-66ad-4d5f-9e2e-6f408bd0c313.png)
![2022-12-08-Code-1670491586](https://user-images.githubusercontent.com/52764066/206411632-eb04e885-cfb5-47f1-a110-12b433c2d978.png)

Steps to Reproduce:

1. Open an empty folder in vscode
2. Create an empty index.js file
3. Create a jsconfig.json with valid properties without specifying a compiler options target.

**Comment by Sun-ZhenXing** [comment]
Same question, when creating vite project:

```bash
pnpm create vite
# config...

cd my-proj
pnpm i
pnpm dev
```

9 Errors in `tsconfig.node.json`:

```json
[{
	"resource": "/d:/WorkSpace/sanic-vue-exam/vue-simple-exam/tsconfig.node.json",
	"owner": "typescript",
	"severity": 8,
	"message": "File 'd:/Pro/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/lib.d.ts' not found.\n  The file is in the program because:\n    Default library for target 'es3'",
	"source": "ts",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 2
}]
```

**Comment by Medkhat** [comment]
> 
I also had this error in `tsconfig.node.json`. 
I restarted TS Server in VSCode and then it fixed.

`cmd + shift + p` 

<img width="336" alt="image" src="https://user-images.githubusercontent.com/42961493/206461940-27b63f8a-9169-4f3f-afc6-f3c84e99cc27.png">

**Comment by blackOw1** [comment]
> I also had this error in `tsconfig.node.json`. I restarted TS Server in VSCode and then it fixed.
> 
> `cmd + shift + p`
> 
> <img alt="image" width="336" src="https://user-images.githubusercontent.com/42961493/206461940-27b63f8a-9169-4f3f-afc6-f3c84e99cc27.png">

I tried this, but it didn't work for me. I thought I was the only one with this issue after updating VS Code.

**Comment by dcarias0388** [comment]
Igual me esta pasando, de la nada salio el problema.

**Comment by sagargulati** [comment]
Same issue on Mac Ventura

**Comment by kevinswarner** [comment]
I am also having this issue on Mac Ventura. Restarting the Typescript server or even restarting VS Code only solves the problem for a few minutes, then it resurfaces.

**Comment by ashmchang** [comment]
Having the same issue. Can anyone share what the lib.d.ts file should look like? I don't think I have one in a recent project to try @blackOw1's solution

**Comment by blackOw1** [comment]
> Having the same issue. Tried the two solutions above and still having the same issue.

Try downloading an older version of VS Code and copy the file from there to the location of the error you are seeing. Then, try saving via the jsonfig.json file.

**Comment by sagargulati** [comment]
Here is what worked for me!

Close the VS Code.

`yarn global add next@latest`
`yarn global add typescript@latest`

Open your project
`rm -rf node_modules package-lock.json`

Re-open your project!

Also, check your **VS Code extension**, Install the official **Javascript & Typescript** [extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) by Microsoft.

I hope this helps!

**Comment by vialoh** [comment]
Thank you @sagargulati. Installing the extension and restarting VSCode worked for me.

**Comment by IvanovRoman** [comment]
Thank you @sagargulati. Installing the extension and restarting VSCode worked for me.

**Comment by Urszula-Molska** [comment]
It worked for me too :)

**Comment by gnuion** [comment]
Added `"target": "esnext"` and now it just works.

```
# tsconfig.node.js
{
  "compilerOptions": {
    "target": "esnext",
    "composite": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "vite.config.ts"
  ]
}
```

**Comment by S-STALWART-S** [comment]
Thanks @sagargulati.  install the extension worked for me

**Comment by segevfiner** [comment]
Same in a `create-vue` app, same cause.

**Comment by HugoRamonPereira** [comment]
I am having this very same problem in the tsconfig.node.json with Vite + React. Default library for target 'es3'

**Comment by HugoRamonPereira** [comment]
I tried all the solutions suggested above but none of it worked. I am using React + Vite + Typescript. 
![Screenshot 2022-12-11 at 22 01 58](https://user-images.githubusercontent.com/44532901/206940888-80174f19-8c1e-407d-8c44-fc253210b057.png)

**Comment by Wellitonzs** [comment]
> Added `"target": "esnext"` and now it just works.
> 
> ```
> # tsconfig.node.js
> {
>   "compilerOptions": {
>     "target": "esnext",
>     "composite": true,
>     "module": "ESNext",
>     "moduleResolution": "Node",
>     "allowSyntheticDefaultImports": true
>   },
>   "include": [
>     "vite.config.ts"
>   ]
> }
> ```
Thanks man. It's worked for me.

**Comment by hbnxgdh** [comment]
Added "target": "ESNext" and now it just works.
{
  "compilerOptions": {
    "target": "ESNext",
    "composite": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "allowSyntheticDefaultImports": true
  },
  "include": ["vite.config.ts"]
}

**Comment by reservecrate** [comment]
@hbnxgdh is a Gigachad

**Comment by CodeThag** [comment]
> I also had this error in `tsconfig.node.json`. I restarted TS Server in VSCode and then it fixed.
> 
> `cmd + shift + p`
> 
> <img alt="image" width="336" src="https://user-images.githubusercontent.com/42961493/206461940-27b63f8a-9169-4f3f-afc6-f3c84e99cc27.png">

This worked for me. Thanks

**Comment by ndhieu2701** [comment]
add "target": "esVersion" with esVersion like es6, es2017, es2020, ...
this worked for me :)

**Comment by daramir** [comment]
Seems fixed on **Update 1.74.1**. Thanks!

**Comment by andrei22b** [comment]
> 

works with the latest update

**Comment by andrei22b** [comment]
@daramir i confirm, the latest update works

**Comment by Asitha-Karnatakam** [comment]
Problem fixed on Update@1.74.1

**Comment by iksflow** [comment]
I fixed this problem too on 1.74.1
